### PR TITLE
CooMatrix::reserve added

### PR DESCRIPTION
**Pull request by chammika-become**
PR for  #881
- Added doc tests as well. 

Note: I've noticed missing doc-tests in some of the APIs. Can add them in future PRs if it helps the documentation.

### CODE REVIEW

File: nalgebra-sparse/src/coo.rs [hunk 1]
```diff
@@ -130,6 +130,30 @@ impl<T> CooMatrix<T> {
             .map(|((i, j), v)| (*i, *j, v))
     }
 
+    /// Reserves capacity for COO matrix by at least `additional` elements.
+    ///
+    /// This increase the capacities of triplet holding arrays by reserving more space to avoid
+    /// frequent reallocations, in `push` operations.
```
Comment (Andlon) [inline nalgebra-sparse/src/coo.rs:136]: maybe remove the comma here? ("... frequent reallocations in `push` operations)

File: nalgebra-sparse/src/coo.rs [hunk 2]
```diff
@@ -130,6 +130,30 @@ impl<T> CooMatrix<T> {
...
+    ///
+    /// ## Panics
+    ///
+    /// Panics if any of the individual allocation of triplet arrays fail.
```
Comment (Andlon) [inline nalgebra-sparse/src/coo.rs:140]: Perhaps "Panics if any of the individual allocations of triplet arrays fails."

File: nalgebra-sparse/src/coo.rs [hunk 3]
```diff
@@ -130,6 +130,30 @@ impl<T> CooMatrix<T> {
...
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// # use nalgebra_sparse::coo::CooMatrix;
+    /// let mut coo = CooMatrix::new(4, 4);
+    /// // Reserver capacity in advance
```
Comment (Andlon) [inline nalgebra-sparse/src/coo.rs:147]: Reserver -> Reserve?

**Review by Andlon — CHANGES_REQUESTED** [review]
Thanks for this PR, @chammika-become! Looks good to me, and exactly as intended in #881. I only noticed a couple of very minor grammatical things in the doc comments (I hope I got it correct myself, my English is not flawless).

I think the docs would definitely benefit from some additional doc tests, yes! Would be happy to see additional PRs if you would like to work on this :)

**Comment by chammika-become** [comment]
Thank you for the corrections. I will pay more attention to grammar next time.

**Comment by Andlon** [comment]
No worries, as I said they were super minor, and it wouldn't really have been necessary to fix them either, because the meaning was clear. I just prefer fixing them up front, because nobody will bother to fix it later!

**Review by Andlon — APPROVED** [review]

**Comment by sebcrozet** [comment]
Thanks!

**Pull request merged**